### ISSUE TITLE
i3bar: Error: status_command process exited unexpectedly (exit 1)

### ISSUE THREAD

**Issue by karma0** [i3/i3#3015]
Output of `i3 --moreversion 2>&- || i3 --version`:

    Binary i3 version:  4.14.1 (2017-09-24) © 2009 Michael Stapelberg and contributors
    Running i3 version: 4.14.1 (2017-09-24) (pid 516) abort…)
    Loaded i3 config: /home/karma0/.i3/config (Last modified: Fri 06 Oct 2017 11:28:25 AM MDT, 1074 seconds ago)
    
    The i3 binary you just called: /usr/bin/i3
    The i3 binary you are running: i3

URL to a logfile as per https://i3wm.org/docs/debugging.html:

    https://logs.i3wm.org/logs/5630721452408832.bz2

**What I did:**

You can view my i3 config [here](https://github.com/karma0/i3-config/tree/master/.i3).  Pay particular attention to the `bar` section.  The `status_command` is `~/.i3/status.py laptop` and it works fine when executed from the command line.

**What I saw:**

This configuration is used across all of my systems, and works fine on all but one.  On the system that doesn't work as expected, the i3bar reports:

    Error: status_command process exited unexpectedly (exit 1)

In following the _Debugging i3bar_ steps [on this page](https://i3wm.org/docs/debugging.html), the bar starts working with only a warning that _i3bar should be started by i3_ .  If I `reload` i3 or reboot, the error returns.

The other `bar` parameters seem to work, as I only seem to receive the error with the `status_command` entry.

**What I expected instead:**

This is the output from the debugging step above (or, more specifically, when I run `killall i3bar; i3bar --bar_id=bar-0`):

```
[libi3] ../i3-4.14.1/../i3-4.14.1/libi3/font.c Using Pango font Source Code Pro, size 7
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1329] Calculated font height: 13
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1466] Setting bar_fg = #FFFFFF as _NET_SYSTEM_TRAY_COLORS
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1088] Received X11 error, sequence 0x25, error_code = 3
[../i3-4.14.1/../i3-4.14.1/i3bar/src/ipc.c:215] Got data!
[../i3-4.14.1/../i3-4.14.1/i3bar/src/ipc.c:66] Clearing old output configuration...
[../i3-4.14.1/../i3-4.14.1/i3bar/src/ipc.c:73] Parsing outputs JSON...
[../i3-4.14.1/../i3-4.14.1/i3bar/src/ipc.c:75] Reconfiguring windows...
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1668] Destroying window for output xroot-0
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1673] Creating window for output eDP-1
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1373] Initializing system tray functionality
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1466] Setting bar_fg = #FFFFFF as _NET_SYSTEM_TRAY_COLORS
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1940] Drawing bars...
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1950] Output xroot-0 inactive, skipping...
[../i3-4.14.1/../i3-4.14.1/i3bar/src/ipc.c:215] Got data!
[../i3-4.14.1/../i3-4.14.1/i3bar/src/ipc.c:57] Got subscribe reply: {"success":true}
[../i3-4.14.1/../i3-4.14.1/i3bar/src/ipc.c:215] Got data!
[../i3-4.14.1/../i3-4.14.1/i3bar/src/ipc.c:46] Got workspace data!
[../i3-4.14.1/../i3-4.14.1/i3bar/src/workspaces.c:140] Got workspace canonical: 1:１, name: '１', name_width: 8, glyphs: 1
[../i3-4.14.1/../i3-4.14.1/i3bar/src/workspaces.c:140] Got workspace canonical: 11: , name: ' ', name_width: 10, glyphs: 2
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1940] Drawing bars...
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1950] Output xroot-0 inactive, skipping...
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1967] Drawing button for WS １ at x = 0, len = 8
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1967] Drawing button for WS   at x = 19, len = 10
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:683] _NET_SYSTEM_TRAY_OPCODE received
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:728] xembed format = 32, len = 2
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:730] xembed version = 1
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:731] xembed flags = 1
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:741] X window 00e00083 requested docking
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:762] Found tray_output assignment on primary output eDP-1.
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:843] Mapping dock client
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1940] Drawing bars...
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1950] Output xroot-0 inactive, skipping...
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1967] Drawing button for WS １ at x = 0, len = 8
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1967] Drawing button for WS   at x = 19, len = 10
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:683] _NET_SYSTEM_TRAY_OPCODE received
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:728] xembed format = 32, len = 2
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:730] xembed version = 1
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:731] xembed flags = 1
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:741] X window 00e00087 requested docking
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:762] Found tray_output assignment on primary output eDP-1.
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:843] Mapping dock client
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1940] Drawing bars...
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1950] Output xroot-0 inactive, skipping...
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1967] Drawing button for WS １ at x = 0, len = 8
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1967] Drawing button for WS   at x = 19, len = 10
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:897] MapNotify for window = 00e00083, event = 00e00083
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:903] checking output eDP-1
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:909] Tray client mapped (window ID 00e00083). Adjusting tray.
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:663] Configuring tray window 00e00083 to x=1903
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1940] Drawing bars...
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1950] Output xroot-0 inactive, skipping...
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1967] Drawing button for WS １ at x = 0, len = 8
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1967] Drawing button for WS   at x = 19, len = 10
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:897] MapNotify for window = 00e00087, event = 00e00087
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:903] checking output eDP-1
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:909] Tray client mapped (window ID 00e00087). Adjusting tray.
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:663] Configuring tray window 00e00087 to x=1903
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:663] Configuring tray window 00e00083 to x=1886
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1940] Drawing bars...
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1950] Output xroot-0 inactive, skipping...
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1967] Drawing button for WS １ at x = 0, len = 8
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1967] Drawing button for WS   at x = 19, len = 10
[../i3-4.14.1/../i3-4.14.1/i3bar/src/ipc.c:215] Got data!
[../i3-4.14.1/../i3-4.14.1/i3bar/src/ipc.c:46] Got workspace data!
[../i3-4.14.1/../i3-4.14.1/i3bar/src/workspaces.c:140] Got workspace canonical: 1:１, name: '１', name_width: 8, glyphs: 1
[../i3-4.14.1/../i3-4.14.1/i3bar/src/workspaces.c:140] Got workspace canonical: 11: , name: ' ', name_width: 10, glyphs: 2
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1940] Drawing bars...
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1950] Output xroot-0 inactive, skipping...
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1967] Drawing button for WS １ at x = 0, len = 8
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1967] Drawing button for WS   at x = 19, len = 10
[../i3-4.14.1/../i3-4.14.1/i3bar/src/child.c:427] Detecting input type based on buffer *{"version": 1, "click_events": true}
[
*
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1940] Drawing bars...
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1950] Output xroot-0 inactive, skipping...
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1967] Drawing button for WS １ at x = 0, len = 8
[../i3-4.14.1/../i3-4.14.1/i3bar/src/xcb.c:1967] Drawing button for WS   at x = 19, len = 10
[../i3-4.14.1/../i3-4.14.1/i3bar/src/child.c:300] copying statusline_buffer to statusline_head
[../i3-4.14.1/../i3-4.14.1/i3bar/src/child.c:304] dumping statusline:
[../i3-4.14.1/../i3-4.14.1/i3bar/src/child.c:311] end of dump
[../i3-4.14.1/../i3-4.14.1/i3bar/src/child.c:300] copying statusline_buffer to statusline_head
[../i3-4.14.1/../i3-4.14.1/i3bar/src/child.c:304] dumping statusline:
[../i3-4.14.1/../i3-4.14.1/i3bar/src/child.c:307] full_text = Start pomodoro
[../i3-4.14.1/../i3-4.14.1/i3bar/src/child.c:308] short_text = (null)
[../i3-4.14.1/../i3-4.14.1/i3bar/src/child.c:309] color = #2ECCFA
[../i3-4.14.1/../i3-4.14.1/i3bar/src/child.c:307] full_text = ♪100%
[../i3-4.14.1/../i3-4.14.1/i3bar/src/child.c:308] short_text = (null)
[../i3-4.14.1/../i3-4.14.1/i3bar/src/child.c:309] color = #E6E1CF
[../i3-4.14.1/../i3-4.14.1/i3bar/src/child.c:307] full_text = 192.168.104.206/24
[../i3-4.14.1/../i3-4.14.1/i3bar/src/child.c:308] short_text = (null)
[../i3-4.14.1/../i3-4.14.1/i3bar/src/child.c:309] color = #00FF00
[../i3-4.14.1/../i3-4.14.1/i3bar/src/child.c:307] full_text = enp3s0f1: DOWN
[../i3-4.14.1/../i3-4.14.1/i3bar/src/child.c:308] short_text = (null)
[../i3-4.14.1/../i3-4.14.1/i3bar/src/child.c:309] color = #FF0000
[../i3-4.14.1/../i3-4.14.1/i3bar/src/child.c:307] full_text = _______________:04,03,04,03
[../i3-4.14.1/../i3-4.14.1/i3bar/src/child.c:308] short_text = (null)
[../i3-4.14.1/../i3-4.14.1/i3bar/src/child.c:309] color = #bece51
[../i3-4.14.1/../i3-4.14.1/i3bar/src/child.c:307] full_text = ^██        
[../i3-4.14.1/../i3-4.14.1/i3bar/src/child.c:308] short_text = (null)
[../i3-4.14.1/../i3-4.14.1/i3bar/src/child.c:309] color = #d6ce4b
[../i3-4.14.1/../i3-4.14.1/i3bar/src/child.c:307] full_text = Fri 6 Oct 12:07:15 PM W40
[../i3-4.14.1/../i3-4.14.1/i3bar/src/child.c:308] short_text = (null)
[../i3-4.14.1/../i3-4.14.1/i3bar/src/child.c:309] color = #E6E1CF
[../i3-4.14.1/../i3-4.14.1/i3bar/src/child.c:311] end of dump
```

**In Summary**

* The issue appears only on one machine of all of my machines.
* The issue appears when i3 runs i3bar.
* The bar, `status_command` executes without error when executed by hand.
 
Is there a better facility to debug this issue?

**Comment by karma0** [comment]
It seems the issue had more to do with my Python environment than i3.  With the `status_command` line I was able to do the following:

```
status_command ~/.i3/status.py laptop 2> ~/.i3bar.out
```

The contents of the log file showed Python module errors.  `pyenv` is what I'm using to manage things on my system, so I needed to use the Arch package manager to install the missing dependencies since `pyenv` doesn't get initialized until after ZSH starts (i3 is started from `.xinitrc`).

After that, everything worked.
